src/main/java/data/DatabaseController.java

```diff
@@ -0,0 +1,1032 @@
+package data;
+
+import java.sql.*;
+import java.util.ArrayList;
+import java.util.Collection;
+
+import pathfinding.Node;
+import pathfinding.ConcreteNode;
+
+import static java.sql.DriverManager.println;
+
+public class DatabaseController {
+
+	//private static String dbURL = "jdbc:derby://localhost:1527/myDB;create=true;user=me;password=mine";
+	static final String DB_URL = "jdbc:derby:FHAlpha;create=true";
+	private static String providerTable = "Provider";
+	private static String nodeTable = "Node";
+	private static String officeTable = "Office";
+	private static String neighborTable = "Neighbor";
+	private static String floorTable = "Floor";
+	// jdbc Connection
+	private static Connection connection = null;
+	private static Statement stmt = null;
+	private static ArrayList<Node> nodeList = new ArrayList<Node>();
+	private static ArrayList<Provider> providerList = new ArrayList<Provider>();
+	private static ArrayList<Floor> floorList = new ArrayList<Floor>();
+
+	//TODO: Remove main and properly initialize connections/tabes from elsewhere
+	public static void main(String[] args) {
+		createConnection();
+
+		//initialize tables
+		initializeProviderTable();
+		initializeFloorTable();
+		initializeNodeTable();
+		initializeOfficeTable();
+		initializeNeighborTable();
+
+		shutdown();
+	}
+
+	public static void droptablesForShittyTesting(){
```

Comment:
This should probably be removed, or moved elsewhere so that it's not publicly visible. Testing code and actual code should ideally be separated.